### PR TITLE
SOLR-14569: Configuring a shardHandlerFactory on the /select requestHandler results in HTTP 401 when searching on alias in secured Solr

### PR DESCRIPTION
**Pull request by markrmiller**
Security setups do not work correctly with per request handler shardHandlerFactory.

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/core/CoreContainer.java [hunk 1]
```diff
@@ -737,10 +746,10 @@ public void load() {
 
     zkSys.initZooKeeper(this, cfg.getCloudConfig());
     if (isZooKeeperAware()) {
-      pkiAuthenticationPlugin = new PKIAuthenticationPlugin(this, zkSys.getZkController().getNodeName(),
+      pkiAuthenticationSecurityBuilder = new PKIAuthenticationPlugin(this, zkSys.getZkController().getNodeName(),
           (PublicKeyHandler) containerHandlers.get(PublicKeyHandler.PATH));
       // use deprecated API for back-compat, remove in 9.0
-      pkiAuthenticationPlugin.initializeMetrics(solrMetricsContext, "/authentication/pki");
+      pkiAuthenticationSecurityBuilder.initializeMetrics(solrMetricsContext, "/authentication/pki");
```
Comment (epugh) [inline solr/core/src/java/org/apache/solr/core/CoreContainer.java:752]: If this is for 9, is it possible to remove this?   Or is removing this mean other things need to happen that are orthogonal to this PR?

Comment (markrmiller) [reply solr/core/src/java/org/apache/solr/core/CoreContainer.java:752]: Yeah, this is 9x, but I don't really see what that comment refers too - I don't see that initializeMetrics is deprecated in the super class chain or what this would be changed to.

**Comment by anshumg** [comment]
Hey @markrmiller, is this still a draft ?

**Comment by markrmiller** [comment]
No should be good if you want to take a look. Just marked it as draft because I had converted it from 8x and didn't know if a fail I was seeing consistently was somehow related, but it wasn't.

**Comment by anshumg** [comment]
I'll take a look, thanks 👍🏽

File: solr/core/src/test/org/apache/solr/security/TestAuthWithShardHandlerFactoryOverride.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Similar to BasicAuthOnSingleNodeTest, but using a different configset, to test SOLR-14569
+ */
+public class TestAuthWithShardHandlerFactoryOverride extends SolrCloudAuthTestCase {
+
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
```
Comment (anshumg) [inline solr/core/src/test/org/apache/solr/security/TestAuthWithShardHandlerFactoryOverride.java:38]: We can get rid of this as we don't log anything.

File: solr/core/src/test/org/apache/solr/security/TestAuthWithShardHandlerFactoryOverride.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Similar to BasicAuthOnSingleNodeTest, but using a different configset, to test SOLR-14569
+ */
+public class TestAuthWithShardHandlerFactoryOverride extends SolrCloudAuthTestCase {
```
Comment (anshumg) [inline solr/core/src/test/org/apache/solr/security/TestAuthWithShardHandlerFactoryOverride.java:36]: Let's rename this to have a `Test` suffix instead? Trying to follow the more used convention.

**Review by anshumg — APPROVED** [review]
Left a couple of minor nit-picks, but over all the code looks good.

Tested this out manually by enabling rule based auth followed by:
1. Create multi-shard collection w/ a custom shard handler factory
2. Send exampledoc updates via bin/post 
3. Query the docs and get back response
4. Create an alias pointing to the collection
5. Send updates to the alias
6. Query the alias for documents and verify the response

**Comment by markrmiller** [comment]
Thanks Anshum, made those changes.
